cpp/src/arrow/compute/row/compare_internal_avx2.cc

```diff
@@ -240,12 +206,20 @@ uint32_t KeyCompare::CompareBinaryColumnToRowHelper_avx2(
             _mm256_loadu_si256(reinterpret_cast<const __m256i*>(left_to_right_map) + i);
       }
 
-      __m256i offset_right =
-          _mm256_mullo_epi32(irow_right, _mm256_set1_epi32(fixed_length));
-      offset_right = _mm256_add_epi32(offset_right, _mm256_set1_epi32(offset_within_row));
-
-      reinterpret_cast<uint64_t*>(match_bytevector)[i] =
-          compare8_fn(rows_left, rows_right, i * unroll, irow_left, offset_right);
+      __m256i irow_right_lo = _mm256_cvtepi32_epi64(_mm256_castsi256_si128(irow_right));
```

Comment:
May you add a short common to explain those non-trivial SIMD code sequences?

Reply:
Sure, I'll add some comments about them. Update soon.

Reply:
Done. Added comments for the SIMD code I think need explanation.